local-cli/runIOS/runIOS.js

```diff
@@ -129,7 +133,15 @@ function buildProject(xcodeProject, udid, scheme) {
     '-derivedDataPath', 'build',
   ];
   console.log(`Building using "xcodebuild ${xcodebuildArgs.join(' ')}"`);
-  child_process.spawnSync('xcodebuild', xcodebuildArgs, {stdio: 'inherit'});
+  var buildProcess = child_process.spawnSync('xcodebuild', xcodebuildArgs, {stdio: 'pipe'});
+  var buildOutput = buildProcess.stdout.toString();
+  console.log(buildOutput);
+  //FULL_PRODUCT_NAME is the actual file name of the app, which actually comes from the Product Name in the build config, which does not necessary match a scheme name,  example output line: export FULL_PRODUCT_NAME="Super App Dev.app"
+  let productNameMatch = /(export FULL_PRODUCT_NAME=)("*)(.+)(.app)/.exec(buildOutput);
+  if (productNameMatch && productNameMatch.length && productNameMatch.length > 3)
+  {
+    return productNameMatch[3];//0 is the full match, 1 is 'export FULL_PRODUCT_NAME=', 2 is possible double quote, 3 is app name and 4 is '.app'
```

Comment:
You don't need to use `()` for everything, just put it around the part of the input string you care about.

`const productNameMatch = /export FULL_PRODUCT_NAME="?(.+)\.app/.exec(buildOutput);`